bin/wiki_entity_linking/entity_linker_evaluation.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import random
+
+from collections import defaultdict
+from tqdm import tqdm
+from typing import Dict, NamedTuple
+
+logger = logging.getLogger(__name__)
+
+
+class BaselineResults(NamedTuple):
+    counts_dict: Dict
+    random_recall: float
+    random_precision: float
+    random_accuracy_by_label: Dict
+    prior_recall: float
+    prior_precision: float
+    prior_accuracy_by_label: Dict
+    oracle_recall: float
+    oracle_precision: float
+    oracle_accuracy_by_label: Dict
+
+    def report_accuracy(self, model: str) -> str:
+        model_str = model.title()
+        recall = getattr(self, "{}_recall".format(model))
+        precision = getattr(self, "{}_precision".format(model))
+        return ("{}: ".format(model_str) +
+                "Recall = {} | ".format(round(recall, 3)) +
+                "Precision = {}".format(round(precision, 3)))
+
+
+def measure_performance(dev_data, kb, el_pipe):
+    baseline_accuracies = measure_baselines(
+        dev_data, kb
+    )
+    print("dev counts:", sorted(baseline_accuracies.counts_dict.items(), key=lambda x: x[0]))
+
+    logger.info(baseline_accuracies.report_accuracy("random"))
+    logger.info(baseline_accuracies.report_accuracy("prior"))
+    logger.info(baseline_accuracies.report_accuracy("oracle"))
+
+    # using only context
+    el_pipe.cfg["incl_context"] = True
+    el_pipe.cfg["incl_prior"] = False
+    dev_precision_context, dev_recall_context, dev_acc_cont_d = measure_acc(dev_data, el_pipe)
+    print("dev precision context avg:", round(dev_precision_context, 3))
+    print("dev recall context avg:", round(dev_recall_context, 3))
+
+    # measuring combined accuracy (prior + context)
+    el_pipe.cfg["incl_context"] = True
+    el_pipe.cfg["incl_prior"] = True
+    dev_precision_combo, dev_recall_combo, dev_acc_combo_d = measure_acc(dev_data, el_pipe)
+    print("dev precision combo avg:", round(dev_precision_combo, 3))
+    print("dev recall combo avg:", round(dev_recall_combo, 3))
+
+
+def measure_acc(data, el_pipe=None, error_analysis=False):
```

Comment:
The `error_analysis` parameter is not doing anything anymore because that part of the code was removed. I agree it was kind of "hacky" to put it in the `measure_acc` method, but it was useful to me when debugging :p either way, if we want to leave it out, this parameter can go too.